src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIteratorExtensions.cs

```diff
@@ -90,6 +90,19 @@ public static string GetAsciiStringEscaped(this ReadCursor start, ReadCursor end
             return sb.ToString();
         }
 
+        public static string GetAsciiStringEscaped(this Span<byte> span)
```

Comment:
@davidfowl GetAsciiStringEscaped needs a maxChars parameter like before. Otherwise the error for an HTTPS connection to and HTTP endpoint that you suggested becomes over eight thousand characters.